src/app/services/study-set-firebase.service.ts

```diff
@@ -17,7 +17,10 @@ export class StudySetFirebaseService extends StudySetService{
 
   override getStudySet(id: string): Observable<StudySetData> {
     return defer(() => from(getDoc(doc(this.firestore, 'study-sets', id)) as Promise<DocumentSnapshot>))
-      .pipe(map((docSnap: DocumentSnapshot) => docSnap.data() as StudySetModel))
+      .pipe(
+        map((docSnap: DocumentSnapshot) => docSnap.data() as StudySetModel),
+        tap(sSet => sSet.id = id) // reset the id since it wasn't returned
```

Comment:
I'm not sure why we would need to reset the id of the study set. They should already theoretically be the same if the id is how we fetched the document.